### PR TITLE
Initial esp32p4 support with select

### PR DESCRIPTION
**Pull request by MabezDev**
Just for my reference, don't merge :)

### CODE REVIEW

File: probe-rs/src/architecture/riscv/communication_interface.rs [hunk 1]
```diff
@@ -457,6 +457,11 @@ impl<'state> RiscvCommunicationInterface<'state> {
         // Reset error bits from previous connections
         self.dtm.clear_error_state()?;
 
+        // enable the debug module
```
Comment (MabezDev) [inline probe-rs/src/architecture/riscv/communication_interface.rs:460]: From what I recall, this move of the code is load bearing

File: probe-rs/src/config/target.rs [hunk 2]
```diff
@@ -21,16 +22,7 @@ use super::{
     },
     Core, MemoryRegion, RawFlashAlgorithm, RegistryError, TargetDescriptionSource,
 };
-use crate::architecture::{
```
Comment (MabezDev) [inline probe-rs/src/config/target.rs:24]: Rebase gone awry, ignore

File: probe-rs/src/architecture/riscv/communication_interface.rs (hunk 1 above)
Comment (bugadani) [reply probe-rs/src/architecture/riscv/communication_interface.rs:460]: We can land these modifications outside of the P4 support anyway if we can give proper justification to them. Which I think we can, turning on the DM seems like a logical first step 😅

File: probe-rs/targets/esp32p4.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+          ir_len: 5
+        - name: hp
+          ir_len: 5
+    default_binary_format: idf
+    cores:
+      - name: main
+        type: riscv
+        core_access_options: 
+          !Riscv
+            jtag_tap: 1
```
Comment (bugadani) [inline probe-rs/targets/esp32p4.yaml:20]:
```suggestion
            jtag_tap: 1
            hart_id: 0
```

While implicitly there already, it's a good example for the second core later

File: probe-rs/targets/esp32p4.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        range:
+          start: 0x0
+          end: 0x4000000
+        is_boot_memory: true
+        cores:
+          - main
+      - !Nvm
+        range:
+          start: 0x40000000
+          end: 0x43FFFFFF
```
Comment (bugadani) [inline probe-rs/targets/esp32p4.yaml:32]: +1 please

File: probe-rs/targets/esp32p4.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        is_boot_memory: false
+        cores:
+          - main
+      - !Ram
+        range:
+          start: 0x4FF00000
+          end: 0x4FFBFFFF
```
Comment (bugadani) [inline probe-rs/targets/esp32p4.yaml:39]: +1 please

**Pull request closed without merging**